src/serac/physics/tests/solid_nonlinear_solve.cpp

```diff
@@ -5,48 +5,269 @@
 #include "mfem.hpp"
 
+#include "serac/physics/materials/liquid_crystal_elastomer.hpp"
+
 #include "serac/mesh/mesh_utils.hpp"
 #include "serac/physics/state/state_manager.hpp"
 #include "serac/physics/materials/solid_material.hpp"
 #include "serac/serac_config.hpp"
+#include "serac/infrastructure/profiling.hpp"
+#include "serac/infrastructure/terminator.hpp"
 
 using namespace serac;
 
-void functional_solid_test_nonlinear_buckle()
+std::string mesh_path = ".";
+
+enum Prec
+{
+  JACOBI,
+  STRUMPACK,
+  CHOLESKI,
+  LU,
+  MULTIGRID,
+  PETSC_MULTIGRID
+};
+
+enum NonlinSolve
 {
-  MPI_Barrier(MPI_COMM_WORLD);
+  NEWTON,
+  LINESEARCH,
+  CRITICALPOINT,
+  TRUSTREGION
+};
+
+NonlinSolve nonlinSolve = NonlinSolve::TRUSTREGION;
+Prec        prec        = Prec::JACOBI;
+
+auto get_opts(int max_iters, double abs_tol = 1e-9)
+{
+  serac::NonlinearSolverOptions nonlinear_options{
+      .nonlin_solver = NonlinearSolver::TrustRegion,
+      //.nonlin_solver = NonlinearSolver::NewtonLineSearch, //LineSearch,
+      //.nonlin_solver = NonlinearSolver::PetscNewton,  //LineSearch,
+      //.nonlin_solver = NonlinearSolver::PetscNewtonCriticalPoint, // breaks for snap_cell
+      .relative_tol               = abs_tol,
+      .absolute_tol               = abs_tol,
+      .min_iterations             = 1,
+      .max_iterations             = 2000,
+      .max_line_search_iterations = 20,
+      .print_level                = 1};
+
+  // best for critical point newton: ls = PetscGMRES, petsc_preconditioner = PetscPCType::LU;
+  serac::LinearSolverOptions linear_options = {.linear_solver = LinearSolver::CG,
+                                               //.linear_solver  = LinearSolver::PetscGMRES,
+                                               //.linear_solver  = LinearSolver::PetscCG,
+                                               .preconditioner = Preconditioner::HypreJacobi,
+                                               //.preconditioner = Preconditioner::Petsc,
+                                               //.petsc_preconditioner = PetscPCType::JACOBI,
+                                               //.petsc_preconditioner = PetscPCType::JACOBI_ROWMAX,
+                                               //.petsc_preconditioner = PetscPCType::GAMG,
+                                               //.petsc_preconditioner = PetscPCType::HMG, // Zach's prefered
+                                               //.petsc_preconditioner = PetscPCType::LU,
+                                               //.petsc_preconditioner = PetscPCType::CHOLESKY,
+                                               .relative_tol   = 0.7 * abs_tol,
+                                               .absolute_tol   = 0.7 * abs_tol,
+                                               .max_iterations = max_iters,
+                                               .print_level    = 1};
+
+  switch (nonlinSolve) {
+    case NonlinSolve::NEWTON: {
+      printf("using newton solver\n");
```

Comment:
Nitpick: These could be converted to `SLIC_INFO_ROOT` in case we ever change how we alter/redirect slic streams.

Reply:
I like this, changed.